### PR TITLE
refactor: Enhance error handling, message editing and prompt display

### PR DESCRIPTION
**Pull request by Cristhianzl**
This pull request includes several changes to the `src/frontend/src/modals/IOModal/components/chatView/chatMessage` directory, focusing on enhancing error handling, editing messages, and displaying prompts. The most important changes include the addition of new components for error views, editing messages, and displaying prompts, as well as the refactoring of existing code to improve readability and maintainability.

### Enhancements to Error Handling, Editing Messages, and Displaying Prompts:

* **Error Handling:**
  * Added `ErrorView` component to display detailed error messages and solutions. This includes handling different content types and using `Markdown` for rendering error details.

* **Editing Messages:**
  * Added `EditMessage` component to handle the editing of chat messages, utilizing `Markdown` for rendering and supporting various content types.

* **Displaying Prompts:**
  * Added `PromptView` component to display prompts with dynamic content replacement based on chat messages.

### Code Refactoring and Cleanup:

* **Refactoring:**
  * Moved the `convertFiles` function to a separate helper file `convert-files.ts` for better modularity and reusability. [[1]](diffhunk://#diff-66ca28a3ef190006c7fd509851c10309493f54c2a3009342108ba3c9a122a4cfR1-R20) [[2]](diffhunk://#diff-a5c2764955c9d507dd134f3add3fec7958daa183ee41cfabe54f5c5702a483e7L167-L187)
  * Removed unused imports and code from `newChatMessage.tsx` to improve code readability and maintainability.

* **General Improvements:**
  * Added `useTabVisibility` hook to manage scroll behavior based on tab visibility in `newChatMessage.tsx`.

### CODE REVIEW

File: src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/editMessage/index.tsx
Comment (anovazzi1) [inline src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/editMessage/index.tsx:1]: edit message component should not be responsable for rendering html

File: src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/promptView/index.tsx
Comment (anovazzi1) [inline src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/promptView/index.tsx:1]: I believe this is not used anymore we can delete this, please check with @ogabrielluiz

File: src/frontend/src/modals/IOModal/components/chatViewWrapper/index.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+                    : () => {
+                        setOpen(false);
+                      }
+                }
+              />
+            )}
+          </div>
+        ) : (
+          <span className="flex h-full w-full items-center justify-center font-thin text-muted-foreground">
+            Select an IO component to view
```
Comment (anovazzi1) [inline src/frontend/src/modals/IOModal/components/chatViewWrapper/index.tsx:118]: wrapper for chat view should not handle empty state only chat view

**Review by anovazzi1 — CHANGES_REQUESTED** [review]
really like the type files

File: src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/promptView/index.tsx
Comment (ogabrielluiz) [reply src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/promptView/index.tsx:1]: I don't we think we use this anymore.

File: src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/editMessage/index.tsx
Comment (Cristhianzl) [reply src/frontend/src/modals/IOModal/components/chatView/chatMessage/components/editMessage/index.tsx:1]: you are correct!

File: src/frontend/src/modals/IOModal/components/chatViewWrapper/index.tsx (hunk 1 above)
Comment (Cristhianzl) [reply src/frontend/src/modals/IOModal/components/chatViewWrapper/index.tsx]: you are correct!

**Review by anovazzi1 — APPROVED** [review]
lgtm

**Pull request merged**